Functional.Primitives.Extensions/ResultExtensions.cs

```diff
@@ -36,6 +48,38 @@ public static class ResultExtensions
 		public static async Task<Result<TResult, TFailure>> Select<TSuccess, TFailure, TResult>(this Task<Result<TSuccess, TFailure>> result, Func<TSuccess, TResult> select)
 			=> (await result).Select(select);
 
+		public static Result<Option<TResult>, TFailure> SelectIfSome<TSuccess, TFailure, TResult>(this Result<Option<TSuccess>, TFailure> result, Func<TSuccess, TResult> select)
+			=> result.Select(success => success.Match(x => Option.Some(select(x)), Option.None<TResult>));
+
+		public static async Task<Result<Option<TResult>, TFailure>> SelectIfSome<TSuccess, TFailure, TResult>(this Task<Result<Option<TSuccess>, TFailure>> result, Func<TSuccess, TResult> select)
+			=> (await result).SelectIfSome(select);
+
+		public static Task<Result<Option<TResult>, TFailure>> SelectIfSomeAsync<TSuccess, TFailure, TResult>(this Result<Option<TSuccess>, TFailure> result, Func<TSuccess, Task<TResult>> select)
+		{
+			return result.MatchAsync(
+				success => success.MatchAsync(async value => Result.Success<Option<TResult>, TFailure>(Option.Some(await select(value))), () => Task.FromResult(Result.Success<Option<TResult>, TFailure>(Option.None<TResult>()))),
+				failure => Task.FromResult(Result.Failure<Option<TResult>, TFailure>(failure)));
+		}
+
+		public static async Task<Result<Option<TResult>, TFailure>> SelectIfSomeAsync<TSuccess, TFailure, TResult>(this Task<Result<Option<TSuccess>, TFailure>> result, Func<TSuccess, Task<TResult>> select)
+			=> await (await result).SelectIfSomeAsync(select);
+
+		public static Result<Option<TResult>, TFailure> SelectIfSome<TSuccess, TFailure, TResult>(this Result<Option<TSuccess>, TFailure> result, Func<TSuccess, Option<TResult>> select)
```

Comment:
This is the equivalent of a Bind for option, so I'm not sure the name of right. I see you also have BindIfSome. I think we might need to think about the naming a bit more.

Reply:
Yes, the naming could use some cleanup.  Was originally thinking about the name in terms of the `result` being acted on (in this case, I had considered it a `Select` operation because we're not producing a `Result<,>` with the `select` method, we're just transforming the `TSuccess` type).

Reply:
Conversation happened outside the PR, but so far the best we have is (for `Result<Option<TResult>, TFailure>`-returning extension methods:
- `SelectIfSome` for `T, Func<T, TResult>` (`func` matches existing `Select` method signature)
- `BindIfSome` for `T, Func<T, Result<TResult, TFailure>` (`func` matches existing `Bind` method signature)
- `SelectIfSomeOntoOption` for `T, Func<T, Option<TResult>>`
- `BindIfSomeOntoOption` for `T, Func<T, Result<TResult, TFailure>>`